### PR TITLE
Initial attempt at approximate count

### PR DESCRIPTION
**Pull request by marvinchin**
Tested against complete graphs for n = 5 and n = 10. Results seem fairly encouraging.

### CODE REVIEW

File: graphs.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    return make_random_graph(n, 1)
+
+def make_random_graph(n, density):
+    g = defaultdict(list)
+    for i in range(n):
+        for j in range(n):
+            if i == j:
+                continue
+            if random.random() < density:
+                g[i].append(j)
```
Comment (pangjiahao) [inline graphs.py:14]: Should we make this function to return a undirected graph instead? I think we are only working with undirected graphs

File: approx_count_st.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                continue
+            if neighbor in st[v]:
+                # don't include edges in the st
+                continue
+            edge = (v, neighbor)
+            edges.append(edge)
+    random.shuffle(edges)
+    return edges
+
+def approx_count_st(g, rounds, samples):
```
Comment (pangjiahao) [inline approx_count_st.py:39]: This looks great!

File: graphs.py (hunk 1 above)
Comment (marvinchin) [reply graphs.py:14]: Oops, I think I missed the `g[j].append(i)` by accident. Thanks for catching this!

Comment (marvinchin) [reply graphs.py:14]: fixed :)

**Comment by pangjiahao** [comment]
lgtm

**Pull request merged**